PhysicsTools/PatAlgos/python/slimming/applySubstructure_cfi.py

```diff
@@ -0,0 +1,158 @@
+        dropSpecific = cms.string("1"),
+        dropTagInfos = cms.string("1"),
+    )
+
+    
+    ## Establish references between PATified fat jets and subjets using the BoostedJetMerger
+    process.slimmedJetsAK8PFCHSSoftDropPacked = cms.EDProducer("BoostedJetMerger",
+        jetSrc=cms.InputTag("selectedPatJetsAK8PFCHSSoftDrop"),
+        subjetSrc=cms.InputTag("slimmedJetsAK8PFCHSSoftDropSubjets")
+    )
+    
+    addJetCollection(
+        process,
+        labelName = 'CMSTopTagCHS',
+        jetSource = cms.InputTag('cmsTopTagPFJetsCHS'),
+        jetCorrections = ('AK8PFchs', cms.vstring(['L1FastJet', 'L2Relative', 'L3Absolute']), 'None'),
+        btagDiscriminators = ['pfCombinedSecondaryVertexBJetTags', 'pfCombinedInclusiveSecondaryVertexV2BJetTags'],
+        genJetCollection = cms.InputTag('slimmedGenJetsAK8'), 
+        getJetMCFlavour = False #
+        )
+    process.patJetsCMSTopTagCHS.addTagInfos = True
+    process.patJetsCMSTopTagCHS.tagInfoSources = cms.VInputTag(
+        cms.InputTag('caTopTagInfosPAT')
+        )
+    process.selectedPatJetsCMSTopTagCHS.cut = cms.string("pt > 200")
+
+    addJetCollection(
+        process,
+        labelName = 'CMSTopTagCHSSubjets',
+        jetSource = cms.InputTag('cmsTopTagPFJetsCHS','caTopSubJets'),
+        algo = 'AK',  # needed for subjet flavor clustering
+        rParam = 0.8, # needed for subjet flavor clustering        
+        btagDiscriminators = ['pfCombinedSecondaryVertexBJetTags', 'pfCombinedInclusiveSecondaryVertexV2BJetTags'],
+        jetCorrections = ('AK4PFchs', ['L1FastJet', 'L2Relative', 'L3Absolute'], 'None'),
+        genJetCollection = cms.InputTag('slimmedGenJets'), # Using ak4GenJets for matching which is not entirely appropriate
+        explicitJTA = True,  # needed for subjet b tagging
+        svClustering = True, # needed for subjet b tagging
+        fatJets=cms.InputTag('ak8PFJetsCHS'),             # needed for subjet flavor clustering
+        groomedFatJets=cms.InputTag('cmsTopTagPFJetsCHS') # needed for subjet flavor clustering
+
+        )
+
+    process.slimmedJetsCMSTopTagCHSSubjets = cms.EDProducer("PATJetSlimmer",
+        src = cms.InputTag("selectedPatJetsCMSTopTagCHSSubjets"),
+        packedPFCandidates = cms.InputTag("packedPFCandidates"),
+        dropJetVars = cms.string("1"),
+        dropDaughters = cms.string("0"),
+        dropTrackRefs = cms.string("1"),
+        dropSpecific = cms.string("1"),
+        dropTagInfos = cms.string("1"),
+    )
+    
+    ## Establish references between PATified fat jets and subjets using the BoostedJetMerger
+    process.slimmedJetsCMSTopTagCHSPacked = cms.EDProducer("BoostedJetMerger",
+        jetSrc=cms.InputTag("selectedPatJetsCMSTopTagCHS"),
+        subjetSrc=cms.InputTag("slimmedJetsCMSTopTagCHSSubjets")
+    )
+
+
+    process.packedPatJetsAK8 = cms.EDProducer("JetSubstructurePacker",
+            jetSrc = cms.InputTag("selectedPatJetsAK8"),
+            distMin = cms.double(0.8),
```

Comment:
I was following the convention of the TrivialDeltaRMatcher. I don't have a strong preference.